### PR TITLE
Search improvements + better loader

### PR DESCRIPTION
**Pull request by apepper**
Motivation: When someone clicks on "Load more" a spinner should be shown until all informations are available (more items in the load and the status of "hasMore").

See effect by artificially slowing down `localStorageDataConnection#index`:

```ts
await new Promise((resolve) => setTimeout(resolve, 1500))
```

### CODE REVIEW

File: src/Components/DataBatchContext.tsx [hunk 1]
```diff
@@ -43,15 +45,17 @@ export const DataBatchContextProvider = connect(
...
 
-    const key = [
+    const baseKeys = [
       'DataBatchContextProvider',
       content.id(),
       attribute,
       id,
       tag,
       search,
       // limit is intentionally not included in the key. Otherwise the component would show a loading spinner on every "load more" click.
```
Comment (dcsaszar) [inline src/Components/DataBatchContext.tsx:55]: * [ ] FYI maybe by using `dataKey` and `batchKey` we could make the code clear enough to go without the comment.

**Review by dcsaszar — CHANGES_REQUESTED** [review]
Interesting stuff. As this is purely for aesthetic purposes we must be careful with naming. Simpler is better.

**Comment by apepper** [comment]
REVIEW ([d29afcf](https://github.com/Scrivito/scrivito-portal-app/pull/751/commits/d29afcfc18fc09be2a3a8aa581be421b26f9649b))

File: src/Widgets/DataWidget/DataBatchLoadingIndicator.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const DataBatchLoadingIndicator = connect(
     const dataScope = useData()
     const { hasMore } = useContext(DataBatchContext)
 
-    // use side-effects to trigger loading
+    // subscribe to dataScope.take() and hasMore() to trigger Loading
```
Comment (dcsaszar) [inline src/Widgets/DataWidget/DataBatchLoadingIndicator.tsx:11]: * [ ] FYI the "to trigger" part of the comment is still misleading. What we do here is more of "subscribe to dataScope.take() and hasMore() loading".

File: src/Components/DataBatchContext.tsx [hunk 3]
```diff
@@ -28,7 +28,15 @@ export const DataBatchContextProvider = connect(
...
-    const [search, setSearch] = useState('')
+    const [search, setSearchState] = useState('')
+
+    const setSearch = useCallback(
+      (query: string) => {
+        setSearchState(query)
+        setLimit(configuredLimit)
+      },
+      [configuredLimit],
+    )
```
[the remark this answers is not in the document]
Comment (dcsaszar) [reply src/Components/DataBatchContext.tsx]: The commit message is "Reset limit on new searches". That's why my expectation was:

```ts
    useEffect(() => {
      setLimit(configuredLimit) // <-- reset limit
    }, [configuredLimit, query]) // <-- on new search query
```

But I haven't tried it out.

**Review by dcsaszar — APPROVED** [review]
Nice. Two FYI left.

File: src/Components/DataBatchContext.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (apepper) [reply src/Components/DataBatchContext.tsx]: Nice suggestion. It works and results in better code. Thank you!

**Comment by apepper** [comment]
REVIEW ([66c9a34](https://github.com/Scrivito/scrivito-portal-app/pull/751/commits/66c9a3474b32a4c6f974690c3e44ae35e0bf6288))

**Review by dcsaszar — APPROVED** [review]
Nice. In absence of tests I keep my fingers crossed the last commit didn't change subtle edge cases. If we find a bug it will be way easier to spot though.
